packages/samples/headless-react/src/components/results-per-page/results-per-page.fn.tsx

```diff
@@ -0,0 +1,52 @@
+import {useEffect, useState, FunctionComponent} from 'react';
+import {
+  buildResultsPerPage,
+  ResultsPerPage as HeadlessResultsPerPage,
+} from '@coveo/headless';
+import {engine} from '../../engine';
+
+interface ResultsPerPageProps {
+  controller: HeadlessResultsPerPage;
+  options: number[];
+}
+
+export const ResultsPerPage: FunctionComponent<ResultsPerPageProps> = (
+  props
+) => {
+  const {controller, options} = props;
+  const [, setState] = useState(controller.state);
+
+  useEffect(() => controller.subscribe(() => setState(controller.state)), []);
+
+  const isSetTo = (numberOfResults: number) => {
+    return controller.isSetTo(numberOfResults);
+  };
+
+  const set = (numberOfResults: number) => {
```

Comment:
Why do we need these wrapping methods?

Reply:
Hmm, looking back, I can't find the reason why I did this. I recalled trying to stay consistent with something else, but I can't find any other instance of wrapping methods being used like this. I'll remove them.

Reply:
Let's do the same with the class component please,